sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2649,3 +2649,192 @@ case class Sentences(
     copy(str = newFirst, language = newSecond, country = newThird)
 
 }
+
+/**
+ * A function that returns the bitwise AND of two binary strings.
+ * The byte length of the result is the maximum of the byte lengths of the two input binary
+ * strings. If the two input binary strings are of different byte length they aligned according
+ * to their least significant (right-most) bit.  The shorter binary string is semantically
+ * left-padded with zeros.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(bytes1, bytes2) - Returns the bitwise AND of two binary strings.
+  """,
+  examples = """
+    Examples:
+      > SELECT hex(_FUNC_(unhex('AABB'), unhex('11223344')));
+       00002200
+  """,
+  since = "3.3.0",
+  group = "string_funcs")
+case class BitAnd(bytes1: Expression, bytes2: Expression)
+  extends BinaryExpression with ExpectsInputTypes with NullIntolerant {
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(BinaryType, BinaryType)
+
+  override def dataType: DataType = BinaryType
+
+  override def left: Expression = bytes1
+  override def right: Expression = bytes2
+
+  override def nullSafeEval(left: Any, right: Any): Any = {
+    bytes1.dataType match {
+      case BinaryType => ByteArray.bitwiseAnd(left.asInstanceOf[Array[Byte]],
```

Comment:
You don't need to check the input type here since you specified the types in `override def inputTypes ...`

Reply:
Done.